### ISSUE TITLE
Progress sync trying to connect to Wi-Fi when there's no Wi-Fi connection available if auto sync is on.

### ISSUE THREAD

**Issue by nonameVHSM** [koreader/koreader#10807]
* KOReader version: 2023.06.1-67
* Device: Kindle PW3/KT4

#### Issue
Related to my other issue #10806
When you're not connected to a Wi-Fi network because there is none available, and you have "Automatically keep documents in sync" turned on, progress sync will try to connect to Wi-Fi even if there is no available network, which is the intended behavior from what I understand.
This is an issue because when I leave the house and try to read something, progress sync keeps trying to connect to Wi-Fi so when I try to switch a book I'm stuck waiting for it to fail to connect for half a minute, and if I'm reading a pdf or djvu file koreader will just crash. The only way to avoid these issues is to always manually turn off "Automatically keep documents in sync".
I feel like there should be a "leave off" option for "Action when Wi-Fi is off", and show a message like "No connection, please connect to Wi-Fi to sync progress", instead of the current behavior. I want to be able to read outside, and only have auto sync at home without having to turn it on and off manually.
#### Steps to reproduce
1. Walk out of Wi-Fi range (or forget Wi-Fi connection under kindle settings)
2. Have "Automatically keep documents in sync" turned on (Requires "Action when Wi-Fi is off: turn on" under Network)
3. Open an epub or fixed layout document 
4. "Connecting to Wi-Fi..." prompt shows up, and after a while so does "Error connecting to the network"
5. Go back to file browser
6. If it's an epub you'll be stuck with a "Connecting to Wi-Fi..." for half a minute before it goes to the file browser, if it's a fixed layout document it'll crash as described in #10806

[crash.log](https://github.com/koreader/koreader/files/12328765/crash.log)

**Comment by NiLuJe** [comment]
Can't really do much about this one. We either have no control over the wifi backend, or it's mostly blocking, so we have limited options for escape hatches.

And, in the actual "exit document" case, we have *zero* escape hatches: the very fact that auto_sync is gated being "turn on" and not "prompt" is because we *cannot* have UI interaction in that codepath. *at all*.

As for a "leave off" option, that's an extra option that you'll basically have to toggle at will anyway; and with it, if you forget *that one* on, you'll lose auto-sync when you *do* have connectivity available.

**Comment by NiLuJe** [comment]
TL;DR: You get to eat a 45s timeout ;).

**Comment by NiLuJe** [comment]
> c.f., the original PR for the reasoning behind that

Specifically, https://github.com/koreader/koreader/pull/10605#issuecomment-1616654001

**Comment by nonameVHSM** [comment]
> Can't really do much about this one. We either have no control over the wifi backend, or it's mostly blocking, so we have limited options for escape hatches to begin with.
> 
> And, in the actual "closing a document" case described here, we have _zero_ escape hatches: the very fact that auto sync is gated behind "turn on" and not "prompt" is because we _cannot_ have UI interaction in that codepath. _at all_. (c.f., the original PR for the reasoning behind that, but basically, it boils down to the fact that we don't have separate UI/input/event loops, it's all in a single one, so we can't just stop the events (because we need the document and Reader instances, so we can't return from our onCloseDocument handler) but keep the input and UI rolling).
> 
> As for a "leave off" option, that's an extra option that you'll basically have to toggle at will anyway; and with it, if you forget _that one_ on, you'll lose auto-sync when you _do_ have connectivity available with Wi-Fi off, so it'd just be extra complexity to simply move the issue around ;).

I see, I think I understand the issue now, that is unfortunate but thanks for the explanation.

**Comment by NiLuJe** [comment]
I may just have had a terrible, terrible idea to at least being able to cancel that with a tap.

This may not actually pan out, because it's terrible ^^.
